### PR TITLE
Feature/benchmark_evaluator

### PR DESCRIPTION
**Pull request by XuhuiZhou**
Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed
- [ ] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [ ] Ready for code review

### CODE REVIEW

**Comment by ProKil** [comment]
It seems that the pre-commit breaks your lines, so the `type: ignore` is not on the line you meant to ignore.

**Comment by XuhuiZhou** [comment]
Any chance you can quickly fix this?

**Comment by ProKil** [comment]
Hi @XuhuiZhou Can you briefly explain what are the changes made to serialization in the PR description.

**Comment by XuhuiZhou** [comment]
Basically just the rewards, previously I excluded the overall rewards but now I am adding them back

File: examples/benchmark_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        "secret",
+        "social_rules",
+        "financial_and_material_benefits",
+        "goal",
+    ]
+    human_annotations = get_human_annotations(target_model_patterns)
+    human_annotation_dict = human_annotation_to_episodelog(
+        human_annotations, return_model_episodes=False, aggregate=True
+    )
+    re_annotate_list = list(human_annotation_dict.keys())
```
Comment (ProKil) [inline examples/benchmark_evaluator.py:117]: What does `re_annotate_list` mean? Are they re-annotated?

There are two variables: `re_annotate` and `re_evaluate`. I would assume the first is re-annotated by humans and the second is re-evaluated by a model. Is this understanding correct?

File: examples/benchmark_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    aggregate_human_annotations: list[EpisodeLog] = list(human_annotation_dict.values())  # type: ignore
+    # Call the function with the specified parameters
+
+    re_evaluate_episodes: list[EpisodeLog] = EpisodeLog.find(
+        EpisodeLog.tag == tag
+    ).all()  # type: ignore
+    if len(re_evaluate_episodes) < len(re_annotate_list):
```
Comment (ProKil) [inline examples/benchmark_evaluator.py:124]: What is the reason behind this comparison? I guess the reason I don't understand it is I don't understand the naming of these two variables.

File: examples/benchmark_evaluator.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    dimension: str,
+) -> dict[str, float]:
+    # check the data is present
+    with_dimension_list = [
+        int(not isinstance(relevant_episode.rewards[0], float))
+        for relevant_episode in machine_annotations
+    ]
+    assert sum(with_dimension_list) == len(human_annotations), "Data is missing"
+    if dimension == "overall":
+        dimension_scores_agent1_human = [
```
Comment (ProKil) [inline examples/benchmark_evaluator.py:48]: Use a for loop? This is too repetitive?

File: examples/benchmark_evaluator.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    return human_episodes
+
+
+def get_dimension_correlation(
+    human_annotations: list[EpisodeLog],
+    machine_annotations: list[EpisodeLog],
+    dimension: str,
+) -> dict[str, float]:
+    # check the data is present
+    with_dimension_list = [
```
Comment (ProKil) [inline examples/benchmark_evaluator.py:42]: This naming is confusing. Maybe 'episodes_with_valid_rewards'?

File: examples/benchmark_evaluator.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    ["gpt-4", "gpt-3.5-turbo", "gpt-4"],
+    ["gpt-4", "gpt-3.5-turbo", "togethercomputer/llama-2-70b-chat"],
+    ["gpt-4", "togethercomputer/llama-2-70b-chat", "gpt-3.5-turbo"],
+]
+
+
+def get_human_annotations(
+    target_model_patterns: list[list[str]],
+) -> list[AnnotationForEpisode]:
+    human_episodes: list[AnnotationForEpisode] = []
```
Comment (ProKil) [inline examples/benchmark_evaluator.py:27]: Maybe `episodes_with_human_annotation`?

File: sotopia/database/aggregate_annotations.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    return ret_rewards
+
+
+def human_annotation_to_episodelog(
+    human_annotation: list[AnnotationForEpisode],
+    return_model_episodes: bool = False,
+    aggregate: bool = False,
+) -> dict[str, EpisodeLog | tuple[EpisodeLog, EpisodeLog]]:
+    """
+    retrieve related episodes and return the {pk: EpisodeLog}
```
[the remark this answers is not in the document]
Comment (ProKil) [reply sotopia/database/aggregate_annotations.py:34]: And the name of this function indicates that the return value is of type `EpisodeLog`, but the true return type is a dict.

File: sotopia/database/aggregate_annotations.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def aggregate_reasoning(reasonings: List[str]) -> str:
+    return "\n".join(reasonings)
+
+
+def aggregate_rewards(
+    rewards: List[Tuple[float, Dict[str, float]]],
+) -> Tuple[float, Dict[str, float]]:
+    def average_dict(item: List[Dict[str, float]]) -> Dict[str, float]:
```
Comment (ProKil) [inline sotopia/database/aggregate_annotations.py:14]: rename `item` -> `dict_list`?

**Review by ProKil — CHANGES_REQUESTED** [review]
As we want to make our code more usable, we should aim for easy-to-understand naming for variables, functions, and better docstring. The feature introduced by this PR is great, but please improve the naming to make code more self-explanatory.

File: examples/benchmark_evaluator.py (hunk 5 above)
Comment (XuhuiZhou) [reply examples/benchmark_evaluator.py:27]: okay

File: examples/benchmark_evaluator.py (hunk 4 above)
Comment (XuhuiZhou) [reply examples/benchmark_evaluator.py:42]: okay!

File: examples/benchmark_evaluator.py (hunk 3 above)
Comment (XuhuiZhou) [reply examples/benchmark_evaluator.py:48]: okay

File: examples/benchmark_evaluator.py (hunk 1 above)
Comment (XuhuiZhou) [reply examples/benchmark_evaluator.py:117]: That's the list to be re evaluated changed

File: examples/benchmark_evaluator.py (hunk 2 above)
Comment (XuhuiZhou) [reply examples/benchmark_evaluator.py:124]: Now it's easier to understand?

File: sotopia/database/aggregate_annotations.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (XuhuiZhou) [reply sotopia/database/aggregate_annotations.py:34]: updated!

File: sotopia/database/aggregate_annotations.py (hunk 7 above)
Comment (XuhuiZhou) [reply sotopia/database/aggregate_annotations.py:14]: updated!

**Comment by XuhuiZhou** [comment]
> As we want to make our code more usable, we should aim for easy-to-understand naming for variables, functions, and better docstring. The feature introduced by this PR is great, but please improve the naming to make code more self-explanatory.

Thanks for the comments!

**Review by ProKil — APPROVED** [review]

**Pull request merged**